### PR TITLE
Feed posting

### PR DESCRIPTION
**Pull request by Mikunj**
**Changes:**

- Users can now select a GIF and tag media when creating posts.
- Fixed SearchScreen not showing specific results.
- Added MediaModal.
- Added GiphyModal.
- Code refactoring.
- Bug fixes for Android.

### CODE REVIEW

**Comment by Mikunj** [comment]
@vevix @edencakir Any idea on how to get a view to hug the keyboard. So when keyboard isn't show the view is at the bottom of the screen, but as soon as keyboard shows it moves up with it and is basically above the keyboard. I had a look at some things but couldn't figure out how to do it :(

**Comment by vevix** [comment]
Maybe you can make use of https://github.com/APSL/react-native-keyboard-aware-scroll-view as we already depend on this?

File: src/screens/Feed/components/GiphyModal/component.js [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+import { getBestGridItemSpacing } from 'kitsu/common/utils';
+import { ProgressiveImage } from 'kitsu/components/ProgressiveImage';
+import * as colors from 'kitsu/constants/colors';
+import { PostImage } from 'kitsu/screens/Feed/components/PostImage';
+import { scene } from 'kitsu/screens/Feed/constants';
+import { styles } from './styles';
+
+const apiKey = 'l2YStwkAmzBNWuFxe';
+const endpoint = 'https://api.giphy.com/v1/gifs/search?';
+const trending = 'https://api.giphy.com/v1/gifs/trending?';
```
Comment (edencakir) [inline src/screens/Feed/components/GiphyModal/component.js]: We can consider putting these into `config > env` or `constants > app`

File: src/screens/Feed/components/GiphyModal/component.js [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+        <ModalHeader
+          title="Select a GIF"
+          leftButtonTitle="Cancel"
+          leftButtonAction={this.handleCancelPress}
+          rightButtonTitle=""
+        />
+        <View style={{ flex: 1 }}>
+          <View style={styles.searchBoxContainer}>
+            <SearchBox
+              placeholder={'Search for a GIF'}
```
Comment (edencakir) [inline src/screens/Feed/components/GiphyModal/component.js]: Minor thing `"Search for a GIF"`

File: src/screens/Feed/components/GiphyModal/component.js [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+              searchIconOffset={116}
+              style={styles.searchBox}
+              value={this.state.query}
```
Comment (edencakir) [inline src/screens/Feed/components/GiphyModal/component.js]: Extract `query` at line 158

File: src/screens/Feed/components/GiphyModal/component.js [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+              onChangeText={text => this.handleSearchStateChange(text)}
+              onSubmitEditing={() => Keyboard.dismiss()}
```
Comment (edencakir) [inline src/screens/Feed/components/GiphyModal/component.js]: `onSubmitEditing={Keyboard.dismiss}`

File: src/screens/Feed/components/GiphyModal/component.js [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+          rightButtonTitle=""
+        />
+        <View style={{ flex: 1 }}>
+          <View style={styles.searchBoxContainer}>
+            <SearchBox
+              placeholder={'Search for a GIF'}
+              searchIconOffset={116}
+              style={styles.searchBox}
+              value={this.state.query}
+              onChangeText={text => this.handleSearchStateChange(text)}
```
Comment (edencakir) [inline src/screens/Feed/components/GiphyModal/component.js]: `onChangeText={this.handleSearchStateChange}`

File: src/screens/Feed/components/MediaModal/component.js [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+
+    this.setState({ loading: true });
+
+    algoliaIndex.search({ query, page }, (err, content) => {
+      if (!err) {
+        const media = page > 0 ? [...this.state.media, ...content.hits] : content.hits;
+        this.setState({ media });
+      }
+
+      this.setState({ loading: false });
```
Comment (edencakir) [inline src/screens/Feed/components/MediaModal/component.js]: Not sure about this one, I mean, if there are no errors, you'll render the whole screen twice. What about this?

 ```
if (!err) {
    const media = page > 0 ? [...this.state.media, ...content.hits] : content.hits;
    this.setState({ media, loading: false });
} else {
    this.setState({ loading: false });
}
```

or we can just leave it as is

File: src/screens/Feed/components/MediaModal/component.js [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+          rightButtonAction={this.handleDonePress}
+          rightButtonDisabled={isNull(selected)}
+        />
+        <View style={{ flex: 1 }}>
+          <View style={styles.searchBoxContainer}>
+            <SearchBox
+              placeholder={'Search for Media'}
+              searchIconOffset={130}
+              style={styles.searchBox}
+              value={this.state.query}
```
Comment (edencakir) [inline src/screens/Feed/components/MediaModal/component.js]: uhm, same here.

File: src/screens/Feed/components/MediaModal/component.js [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+              onChangeText={text => this.handleSearchStateChange(text)}
+              onSubmitEditing={() => Keyboard.dismiss()}
```
Comment (edencakir) [inline src/screens/Feed/components/MediaModal/component.js]: ++

File: src/screens/Feed/components/MediaModal/styles.js [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+    backgroundColor: colors.lightGrey,
+  },
+  pickerRow: {
+    backgroundColor: '#FFFFFF',
+    padding: 8,
+  },
+  pickerIconCircle: {
+    width: 30,
+    height: 30,
+    borderRadius: 30,
```
Comment (edencakir) [inline src/screens/Feed/components/MediaModal/styles.js]: Is this correct? To achieve full circle border, radius should be `width / 2`

File: src/screens/Feed/pages/PostCreation/CreatePost.js [hunk 10]
```diff
@@ -0,0 +1,274 @@
...
+
+  handleFeedPickerModal = (feedPickerModalIsVisible) => {
+    this.setState({ feedPickerModalIsVisible });
+  }
+
+  handlePressPost = async () => {
+    const { navigation } = this.props;
+    const { gif, media } = this.state;
+    const currentUserId = this.props.currentUser.id;
+    const { content, currentFeed } = this.state;
```
Comment (edencakir) [inline src/screens/Feed/pages/PostCreation/CreatePost.js]: Duplicate (Line 102)

File: src/screens/Feed/pages/PostCreation/CreatePost.js [hunk 11]
```diff
@@ -0,0 +1,274 @@
...
+    return (
+      <KeyboardAvoidingView
+        behavior="padding"
+        style={{ flex: 1, backgroundColor: '#FFFFFF' }}
+      >
+        <View style={{ flex: 1 }}>
+          { /* Error */}
+          {!isEmpty(error) &&
+            <View style={{ padding: 6, backgroundColor: '#CC6549' }}>
+              <Text style={{ color: 'white' }}>
```
Comment (edencakir) [inline src/screens/Feed/pages/PostCreation/CreatePost.js]: Stylesheets for performance!

File: src/screens/Feed/pages/PostCreation/CreatePost.js [hunk 12]
```diff
@@ -0,0 +1,274 @@
...
+                An Error Occurred. {error}
+              </Text>
+            </View>
+          }
+          <PostMeta
+            avatar={(currentUser.avatar && currentUser.avatar.medium) || defaultAvatar}
+            author={currentUser.name}
+            feedTitle={this.state.currentFeed.title}
```
Comment (edencakir) [inline src/screens/Feed/pages/PostCreation/CreatePost.js]: line 102

File: src/screens/Feed/pages/PostCreation/CreatePost.js [hunk 13]
```diff
@@ -0,0 +1,274 @@
...
+            onFeedPillPress={() => this.handleFeedPickerModal(true)}
+          />
+          <ScrollView style={{ flex: 1 }} >
+            <PostTextInput
+              inputRef={(el) => { this.postTextInput = el; }}
+              multiline
+              numberOfLines={0}
```
Comment (edencakir) [inline src/screens/Feed/pages/PostCreation/CreatePost.js]: 🤔

File: src/screens/Feed/pages/PostCreation/CreatePost.js [hunk 14]
```diff
@@ -0,0 +1,274 @@
...
+                }
+              }}
+              onSubmitEditing={() => {
+                if (!this.state.content.endsWith('\n')) {
+                  const content = `${this.state.content}\n`;
+                  this.setState({ content });
+                }
+              }}
+              height={Platform.select({ ios: null, android: (this.state.textInputHeight || 0) })}
+              value={this.state.content}
```
Comment (edencakir) [inline src/screens/Feed/pages/PostCreation/CreatePost.js]: Same here

File: src/screens/Feed/pages/PostCreation/CreatePost.js [hunk 15]
```diff
@@ -0,0 +1,274 @@
...
+                  style={{ margin: 10, marginTop: 5 }}
+                />
+              }
+            </View>
+          </ScrollView>
+        </View>
+        <PickerModal
+          visible={this.state.feedPickerModalIsVisible}
+          data={feedStreams.filter(stream => stream.selectable)}
+          currentPick={this.state.currentFeed}
```
Comment (edencakir) [inline src/screens/Feed/pages/PostCreation/CreatePost.js]: and these, please

File: src/screens/Feed/components/MediaModal/styles.js (hunk 9 above)
Comment (Mikunj) [reply src/screens/Feed/components/MediaModal/styles.js]: Seems to work, i just took it from the feed select one

File: src/screens/Feed/components/MediaModal/component.js [hunk 16]
```diff
@@ -64,12 +64,12 @@ class MediaModal extends PureComponent {
...
+      let results = {};
       if (!err) {
-        const media = page > 0 ? [...this.state.media, ...content.hits] : content.hits;
-        this.setState({ media });
+        media = page > 0 ? [...this.state.media, ...content.hits] : content.hits;
+        results = { media };
       }
-
-      this.setState({ loading: false });
+      this.setState({ ...results, loading: false });
```
Comment (edencakir) [inline src/screens/Feed/components/MediaModal/component.js]: Smooth

**Pull request merged**